openstack-cloud/src/main/java/jenkins/plugins/openstack/compute/JCloudsComputer.java

```diff
@@ -47,7 +47,18 @@ public String getCloudName() {
     @Override
     public HttpResponse doDoDelete() throws IOException {
         setTemporarilyOffline(true, OfflineCause.create(Messages._DeletedCause()));
-        getNode().setPendingDelete(true);
+        final JCloudsSlave potentiallyNullNode = getNode();
+
+        /* I added this null check because of NPE for nodes that were already deleted:
+         * see https://issues.jenkins-ci.org/browse/JENKINS-29152
+         * if you look at the overridden method in AbstractCloudComputer class,
+         * it does this null check as well.
+         */
+        if (potentiallyNullNode != null) {
+            potentiallyNullNode.setPendingDelete(true);
+        } else {
+            LOGGER.severe("getNode() returned Null in doDoDelete() method. Cannot mark a Null node for deletion.");
```

Comment:
looking at the supper-implementation, it seems this is a normal condition (ex: user clicks on "Delete node" while it is being removed already). I doubt we need to have this logged at all.